src/Nethermind/Nethermind.JsonRpc/Modules/Rbuilder/RbuilderRpcModule.cs

```diff
@@ -0,0 +1,162 @@
+
+    public ResultWrapper<byte[]> rbuilder_getCodeByHash(Hash256 hash)
+    {
+        return ResultWrapper<byte[]>.Success(worldStateManager.GlobalStateReader.GetCode(hash));
+    }
+
+    public ResultWrapper<Hash256> rbuilder_calculateStateRoot(BlockParameter blockParam, IDictionary<Address, AccountChange> accountDiff)
+    {
+        BlockHeader? blockHeader = blockFinder.FindHeader(blockParam);
+        if (blockHeader is null)
+        {
+            return ResultWrapper<Hash256>.Fail("Block not found");
+        }
+
+        IOverridableWorldScope worldScope = _overridableWorldScopePool.Get();
+        try
+        {
+            IWorldState worldState = worldScope.WorldState;
+            IReleaseSpec releaseSpec = specProvider.GetSpec(blockHeader);
+            worldState.StateRoot = blockHeader.StateRoot!;
+
+            foreach (KeyValuePair<Address, AccountChange> kv in accountDiff)
+            {
+                Address address = kv.Key;
+                AccountChange accountChange = kv.Value;
+
+                if (accountChange.SelfDestructed)
+                {
+                    worldState.DeleteAccount(address);
+                }
+
+                bool hasAccountChange = accountChange.Balance is not null
+                                        || accountChange.Nonce is not null
+                                        || accountChange.CodeHash is not null
+                                        || accountChange.ChangedSlots?.Count > 0;
+                if (!hasAccountChange) continue;
+
+                if (worldState.TryGetAccount(address, out AccountStruct account))
+                {
+                    // IWorldState does not actually have set nonce or set balance.
+                    // Set, its either this or changing `IWorldState` which is somewhat risky.
+                    if (accountChange.Nonce is not null)
+                    {
+                        worldState.SetNonce(address, accountChange.Nonce.Value);
+                    }
+
+                    if (accountChange.Balance is not null)
+                    {
+                        UInt256 originalBalance = account.Balance;
+                        if (accountChange.Balance.Value > originalBalance)
+                        {
+                            worldState.AddToBalance(address, accountChange.Balance.Value - originalBalance, releaseSpec);
+                        }
+                        else if (accountChange.Balance.Value == originalBalance)
+                        {
+                        }
+                        else
+                        {
+                            worldState.SubtractFromBalance(address, originalBalance - accountChange.Balance.Value, releaseSpec);
+                        }
+
+                        if (worldState.GetBalance(address) != accountChange.Balance.Value)
+                        {
+                            throw new Exception("Balance is not same!? Why?");
```

Comment:
`ResultWrapper<Hash256>.Fail` rather than throwing base `Exception`?

Reply:
This should never happen, so I agree with @asdacap having exception thrown here, that is if we don't remove this code.
I think @asdacap that we should delete this altogether?

Reply:
This is more like a debug actually.

Reply:
Removed